File: qubgrademe-classifygrade/src/validation.py
def validate(modules, marks):
    success = True
    message = ""

    # Check if all modules are empty
    areModulesEmpty = all(x is None or len(x.strip()) == 0 for x in modules)
    if areModulesEmpty:
        success = False
        message = "Please provide some module codes and their respective marks"


    if success:
        for i in range(0,len(modules)):
            # Checking if there is a value present for the current module
            if(modules[i] is not None and len(modules[i].strip()) > 0):
                # Checking if there is a valid mark present for this mark
                if not is_string_positive_integer(marks[i]):
                    success = False
                    message = "Please provide a valid integer for every entered module"
                    break
            else:
                # Checking if a mark has been entered when no module name has been entered
                if is_string_positive_integer(marks[i]):
                    success = False
                    message = "Please provide a module name for all marks entered"
                    break

    return (success, message)

def is_string_positive_integer(num: str) -> bool:
    if isinstance(num, int):
        if (num <= 100 and num >= 0): return True
        else: return False
    if num is not None and num.isdigit():
        if int(num) <= 100 and int(num) >= 0:
            return True
    return False

File: qubgrademe-maxmin/src/validation.php
<?php

function validate($modules, $marks)
{
    $message = "";
    if (allNullOrEmpty($modules))
        $message = "Please provide some module codes and their respective marks";

    if (strlen($message) == 0) {
        for ($i = 0; $i < count($modules); $i++) {
            // If there is a valid module at this index
            if (!isModuleNullOrEmpty($modules[$i])) {
                // Check if the mark provided is also valid
                if (!validIntegerInRange($marks[$i])) {
                    $message =
                        "Please provide a valid integer for every entered module";
                    break;
                }
            }
            // If there is not a valid module at this index
            else {
                // Check if there has been a valid mark provided
                if (validIntegerInRange($marks[$i])) {
                    // If there is, send back an error
                    $message = "Please provide a module name for all marks entered";
                    break;
                }
            }
        }
    }

    return $message;
}

function allNullOrEmpty($array)
{
    $areAllNullEmpty = true;
    foreach ($array as $value) {
        if (!isModuleNullOrEmpty($value)) {
            $areAllNullEmpty = false;
        }
    }
    return $areAllNullEmpty;
}

function isModuleNullOrEmpty($value)
{
    return is_null($value) or strlen(trim($value)) == 0;
}

function validIntegerInRange($value)
{
    if (is_null($value)) return false;
    if (is_numeric($value)) {
        $int_value = intval($value);
        if ($int_value <= 100 and $int_value >= 0) {
            return true;
        }
    }
    return false;
}
